### ISSUE TITLE
JRuby/TruffleRuby cannot run railties tests because of use of fork()

### ISSUE THREAD

**Issue by enebo** [rails/rails#35900]
### Steps to reproduce

In railties subdir:
jruby -S bundle exec rake 

### Expected behavior

Get test results.

### Actual behavior

```text
system ~/work/rails/railties (v6.0.0.beta3) * 1189% jruby -S bundle exec rake test
--- test/app_loader_test.rb
/home/enebo/work/jruby/bin/jruby -w -Itest -Ilib -I../activesupport/lib -I../actionpack/lib -I../actionview/lib -I../activemodel/lib test/app_loader_test.rb
rake aborted!
NotImplementedError: fork is not available on this platform
/home/enebo/work/rails/railties/Rakefile:63:in `block in <main>'
/home/enebo/work/rails/railties/Rakefile:51:in `block in <main>'
/home/enebo/work/jruby/bin/bundle:23:in `<main>'
Tasks: TOP => test => test:isolated
(See full trace by running task with --trace)
```

### System configuration
**Rails version**:  v6.0.0.0beta3 tag (although it has been this way since b342db680bd)

**Ruby version**: JRuby (any version but tested with master 9.2.6.0 can show it)

### Other Notes

If I revert b342db680bd we seem to run tests ok but obviously this is doing full process spawns for MRI.

```text
system ~/work/rails/railties (v6.0.0.beta3) *+|REVERTING 1195% jruby -S bundle exec rake test
/home/enebo/work/jruby/bin/jruby -w -Itest:lib:/home/enebo/work/rails/railties/../activesupport/lib:/home/enebo/work/rails/railties/../actionpack/lib:/home/enebo/work/rails/railties/../actionview/lib:/home/enebo/work/rails/railties/../activemodel/lib test/paths_test.rb
Run options: --seed 63377

# Running:

.....................................

Finished in 0.123570s, 299.4248 runs/s, 566.4794 assertions/s.
37 runs, 70 assertions, 0 failures, 0 errors, 0 skips
/home/enebo/work/jruby/bin/jruby -w -Itest:lib:/home/enebo/work/rails/railties/../activesupport/lib:/home/enebo/work/rails/railties/../actionpack/lib:/home/enebo/work/rails/railties/../actionview/lib:/home/enebo/work/rails/railties/../activemodel/lib test/version_test.rb
Run options: --seed 2749

# Running:

..

Finished in 0.021205s, 94.3158 runs/s, 141.4736 assertions/s.
2 runs, 3 assertions, 0 failures, 0 errors, 0 skips
/home/enebo/work/jruby/bin/jruby -w -Itest:lib:/home/enebo/work/rails/railties/../activesupport/lib:/home/enebo/work/rails/railties/../actionpack/lib:/home/enebo/work/rails/railties/../actionview/lib:/home/enebo/work/rails/railties/../activemodel/lib test/engine_test.rb
Run options: --seed 2120

# Running:

..

Finished in 0.021025s, 95.1263 runs/s, 95.1263 assertions/s.
2 runs, 2 assertions, 0 failures, 0 errors, 0 skips
/home/enebo/work/jruby/bin/jruby -w -Itest:lib:/home/enebo/work/rails/railties/../activesupport/lib:/home/enebo/work/rails/railties/../actionpack/lib:/home/enebo/work/rails/railties/../actionview/lib:/home/enebo/work/rails/railties/../activemodel/lib test/path_generation_test.rb
Run options: --seed 43090

# Running:

.

Finished in 0.339270s, 2.9475 runs/s, 5.8950 assertions/s.
1 runs, 2 assertions, 0 failures, 0 errors, 0 skips
/home/enebo/work/jruby/bin/jruby -w -Itest:lib:/home/enebo/work/rails/railties/../activesupport/lib:/home/enebo/work/rails/railties/../actionpack/lib:/home/enebo/work/rails/railties/../actionview/lib:/home/enebo/work/rails/railties/../activemodel/lib test/json_params_parsing_test.rb
Run options: --seed 31558

# Running:

.

Finished in 0.697242s, 1.4342 runs/s, 7.1711 assertions/s.
1 runs, 5 assertions, 0 failures, 0 errors, 0 skips
...
```

**Comment by deepj** [comment]
Just a bit off-topic, but relative to this. Running Rails tests under alternative Ruby implementations such as JRuby and TrufffleRuby or MRI under Windows would a part of Rails CI. This step would help in health and diversity of Ruby community /cc @rafaelfranca @dhh

**Comment by rafaelfranca** [comment]
It used to be, but after years of alternative Ruby implementations always failing builds for no reason other than bugs in the implementation we removed them from out build matrix. The few that remained in our build matrix were made green by the efforts of the community. All the core contributors of Rails use MRI, so we have no interest of making Rails working with alternative implementations but we welcome contributions from the community.

In summary, if someone want to do the work to make Rails tests run with alternative ruby implementation we are happy to merge, but it is not our priority to do so.

**Comment by guilleiguaran** [comment]
@enebo thank you very mucho for opening this!

Maybe you can help us with a patch making this to run with fork on MRI and fallback to spawn in platforms not supporting fork?

we are happy to review a PR doing that!

**Comment by itsWill** [comment]
Threw up a quick PR here: https://github.com/rails/rails/pull/36111
cc: @enebo

**Comment by enebo** [comment]
@guilleiguaran doh.  Sorry I read your request right while boarding an 11 hour plane ride home and totally spaced this out.  Luckily @itsWill picked this up.  The PR looks great to me and I am running railties tests as we speak.
